### PR TITLE
Fixes on Cross Shunted Model

### PR DESCRIPTION
**Pull request by drkostas**

### CODE REVIEW

**Comment by drkostas** [comment]
I checked every part of the integrated code and I couldn't find anything cross-specific that could create that bug. I tried using embeddings from different stages (eg using stage 1 instead of the last etc) and didn't see any change. Then I trained a few models for 15 epochs as @andreicozma1 suggested and I saw that the cross models have similar visualizations to the non-cross models. I also made some improvements to the bash script for shunted training.

**Comment by andreicozma1** [comment]
> I checked every part of the integrated code and I couldn't find anything cross-specific that could create that bug.
> 
> I tried using embeddings from different stages (eg using stage 1 instead of the last etc) and didn't see any change. Then I trained a few models for 15 epochs as @andreicozma1 suggested and I saw that the cross models have similar visualizations with the ones from the shunted non-cross models on that epoch. So I'm merging that branch for now.
> 
> I also made some improvements to the bash script for shunted training.

Cool, thanks for checking - I'll review the code right now. 
Does that mean that the reconstruction issue would come at a later point in training?

File: train_shunted.sh [hunk 1]
```diff
@@ -1,52 +1,149 @@
...
 # MASK_RATIO='0.50'
 ####################################################
 
+# Replace defaults with arguments provided
+parse_args "$@"
+
+# Print the final arguments
+print_args
+
+exit 0
```
Comment (andreicozma1) [inline train_shunted.sh:145]: did you mean to leave the exit 0 in here?

**Review by andreicozma1 — APPROVED** [review]
Changes look good - I'm currently doing some evals on CrossV2 to determine the best configuration, and I'll raise a few MRs for review

**Comment by drkostas** [comment]
> Changes look good - I'm currently doing some evals on CrossV2 to determine the best configuration, and I'll raise a few MRs for review

At some point we need to review together which of your changes are absolutely important to add to my implementations. I've been trying to transfer everything after reviewing your PRs but it's possible I missed something.

**Comment by maofengggg** [comment]
> 
Thank you Konsta. Just double check, currently, the shunted attention's performance still not good, right? or, we have gotten some progress on the performance? ignore the efficience first, let's see performance, shunted attention + baseline vs baseline and shunted attention + cross vs cross. what's the status?

**Comment by andreicozma1** [comment]
> > Changes look good - I'm currently doing some evals on CrossV2 to determine the best configuration, and I'll raise a few MRs for review
> 
> At some point we need to review together which of your changes are absolutely important to add to my implementations. I've been trying to transfer everything after reviewing your PRs but it's possible I missed something.

Sounds good, let's discuss that on Monday at/after the meeting. Tiny version of CrossV2 seemed successful, but currently also training a Small and a Base version to validate results.

**Pull request merged**